Add test for ProtectedResourceMetadataParsing

yannj-fr opened the pull request:
## Motivation and Context
Add tests cases for insurring ProtectedResourceMetadata validation is correctly done

## How Has This Been Tested?
with pytest

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [x] I have added appropriate error handling
- [x] I have added or updated documentation as needed

Kludex commented on tests/server/fastmcp/auth/test_auth_integration.py:
```diff
@@ -335,11 +363,8 @@ class TestAuthEndpoints:
-        print("Sending request to metadata endpoint")
+
         response = await test_client.get("/.well-known/oauth-authorization-server")
-        print(f"Got response: {response.status_code}")
-        if response.status_code != 200:
-            print(f"Response content: {response.content}")
```
Ah, I see why you added it... 😅

Kludex commented on tests/server/fastmcp/auth/test_auth_integration.py:
```diff
@@ -1201,3 +1224,25 @@ async def test_authorize_invalid_scope(self, test_client: httpx.AsyncClient, reg
+
+        response = await protected_resource_test_client.get("/.well-known/oauth-protected-resource")
+        assert response.status_code == 200
+        metadata = response.json()
+        assert metadata == snapshot(
```
```suggestion
        assert response.json() == snapshot(
```

yannj-fr commented:
@Kludex done with the requested changes

Kludex commented on tests/shared/test_auth.py:
Are the tests you added in here necessary?

yannj-fr replied:
To me it prevents someone to introduce regression in the model description by checking against a realistic payload

Kludex commented:
Maybe we could benefit more if we move the test you added to a new file, so we can actually don't mix bad practices with good ones.

yannj-fr commented:
I will create a separated file, actually could be directly in the test/server/auth/test_proctected_resource.py

Kludex commented on tests/server/auth/test_protected_resource.py:
```diff
@@ -0,0 +1,58 @@
+
+
+@pytest.fixture
+async def protected_resource_test_client(protected_resource_app: Starlette):
```
```suggestion
async def test_client(protected_resource_app: Starlette):
```

yannj-fr commented:
As per your advice, I removed the test that is just on the model, and fixed naming.
Just one question, it is not good practice to have test case in a class?

Kludex commented:
> Just one question, it is not good practice to have test case in a class?

I don't think you get anything from it, it's more often seen in old test suites and when using `unittest` (https://docs.python.org/3/library/unittest.html#basic-example).

I prefer to stick with what modern packages do: not have wrapper classes.
